### ISSUE TITLE
`UP038` documentation outdated wrt. fix safety

### ISSUE THREAD

**Issue by VeckoTheGecko** [astral-sh/ruff#13051]
[Docs mention "Fix is always available."](https://docs.astral.sh/ruff/rules/non-pep604-isinstance/) but it is flagged as unsafe in the code
```
ruff --version
ruff 0.6.1
```

### Code
with `isinstance(2, (int,float))` I get

```
test.py:1:1: UP038 Use `X | Y` in `isinstance` call instead of `(X, Y)`
  |
1 | isinstance(2, (int, float))
  | ^^^^^^^^^^^^^^^^^^^^^^^^^^^ UP038
  |
  = help: Convert to `X | Y`

Found 1 error.
No fixes available (1 hidden fix can be enabled with the `--unsafe-fixes` option).
```

Related to #2923

**Comment by MichaReiser** [comment]
Thanks for opening this issue. I can see how this isn't very clear. 

The documentation's fix availability doesn't consider the fix's safety today, which makes sense because the fix is always available when using `--unsafe-fixes`. But I can see how this isn't very clear, especially with the CLI message that explicitly states that the fix is unavailable. 

@zanieb what's your take on fix availability and fix safety?

**Comment by dhruvmanila** [comment]
Yeah, I think it might be useful to add some reference to fix safety in the rule header. It might also be useful to either provide a different icon or a different color to the fix icon in the rules table (https://docs.astral.sh/ruff/rules/#legend).

**Comment by VeckoTheGecko** [comment]
The deprecation of this rule in #16681 means this can be closed

**Comment by VeckoTheGecko** [comment]
OH. I just realised that the title wasn't accurate anymore after the discussion, so maybe this was closed pre-maturely. @dhruvmanila is your comment https://github.com/astral-sh/ruff/issues/13051#issuecomment-2306315853 still relevant here do you think? And if so, would you mind re-opening (I don't have the perms) and updating the title?

**Comment by AlexWaygood** [comment]
UP038 has now been removed, so this issue is no longer relevant
